spec/DBClientSpec.js

```diff
@@ -0,0 +1,253 @@
+                "name": "test",
+                "address": "test",
+                "city": "test"
+            },
+            "file": "test",
+            "ocr": []
+        }
+        this._client.db.create('menus-test', (err) => {
+            if (err) return done(err);
+            const database = this._client.db.use('menus-test');
+            database.insert(menuEntry, (err, data) => {
+                if (err) return done(err);
+                return done();
+            });
+        });
+    }, jasmineTimeout);
+     afterEach((done) => {
+        this._client.db.destroy('menus-test', (err, data) => {
+            if (err) return done(err);
+            return done();
+        });
+    }, jasmineTimeout);
+     describe('DBClient constructor', () => {
+        it('should throws an error when db param is different than undefined/string', () => {
+            expect(function () { new DBClient(1); })
+                .toThrowError('Wrong type for parameter db. Received number, expected string');
+        });
+         it('should have a null value for _db when db param is empty', () => {
+            let dbInstance = new DBClient();
+            expect(dbInstance._db).toBeNull();
+        });
+         it('should have a string value for _db when db param is string', () => {
+            let isString = true;
+            let dbInstance = new DBClient('menus-test');
+            if (typeof dbInstance._db !== 'string') isString = false;
+            expect(isString).toBe(true);
+        });
+     });
+     describe('DBClient.get', () => {
+        it('should throws a NO_DB_ERROR when not initiating the constructor db param and trying to get the database without using the options.db param', () => {
+            let dbInstance = new DBClient();
+            expect(function () { dbInstance.get(); })
+                .toThrowError('This client has no default databaset set, and one was not provided.');
+        });
+         it('should return an array of valid menus schema when initiating the db menus on the constructor', async () => {
+            let menusType;
+            let dbInstance = new DBClient('menus-test');
+            let jsonValidator = new JSONValidator();
+            let menus = await dbInstance.get({ schemaName: 'menus' });
+            let checkMenus = await jsonValidator.sanitizeList(menus, 'menus');
+            if (typeof checkMenus === 'object') menusType = 'object'
+            expect(menusType).toEqual('object');
+        }, jasmineTimeout);
+         it('should return an array of valid menus schema when not initiating the constructor db param and trying to get the database using the options.db === menus param', async () => {
+            let menusType;
+            let dbInstance = new DBClient();
+            let jsonValidator = new JSONValidator();
+            let menus = await dbInstance.get({ db: 'menus-test', schemaName: 'menus' });
+            let checkMenus = await jsonValidator.sanitizeList(menus, 'menus');
```

Comment:
this is not recommended because if we have a bug in our json validator, this could be unreliable.

If we want to test that the schema returned matches a required schema, we must use our validator directly here (ajv)